### PR TITLE
Mypy issues part 3

### PR DESCRIPTION
**Pull request by dstrain115**
More miscellaenous mypy fixes.

### CODE REVIEW

File: qualtran/bloqs/chemistry/pbc/first_quantization/projectile/select_and_prepare.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ def short_name(self) -> str:
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
         if self.is_adjoint:
             # inverting inequality tests at zero Toffoli.
-            return {}
+            return set()
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/pbc/first_quantization/projectile/select_and_prepare.py:111]: whoops

File: qualtran/bloqs/factoring/mod_mul.py [hunk 2]
```diff
@@ -215,13 +218,13 @@ def _generalize_k(b: Bloq) -> Optional[Bloq]:
     return b
 
 
-@bloq_example(generalizer=(ignore_split_join, ignore_alloc_free, _generalize_k))
+@bloq_example(None, generalizer=(ignore_split_join, ignore_alloc_free, _generalize_k))
```
Comment (mpharrigan) [inline qualtran/bloqs/factoring/mod_mul.py:221]: is this required? Kindof ruins the `~*aesthetic*~`

File: qualtran/bloqs/swap_network/cswap_approx_test.py [hunk 3]
```diff
@@ -45,7 +46,9 @@ def test_approx_cswap_t_count(n):
     assert cswap.t_complexity() == cswap_d.t_complexity()
 
 
-def get_t_count_and_clifford(bc: Dict[Bloq, int]) -> Tuple[int, int]:
+def get_t_count_and_clifford(
+    bc: Dict[Bloq, Union[int, sympy.Expr]]
+) -> Tuple[Union[int, sympy.Expr], Union[int, sympy.Expr]]:
```
Comment (mpharrigan) [inline qualtran/bloqs/swap_network/cswap_approx_test.py:51]: Very optional for this and future PRs, but wanted to bring your attention to `qualtran.resource_counting.symbolic_counting_utils.SymbolicInt` as an alias for `Union[int, sympy.Expr]`. 

Also I might move these symbolic utils to a more easily importable location -- perhaps top-level. But that's todo

File: qualtran/bloqs/factoring/mod_mul.py (hunk 2 above)
Comment (dstrain115) [reply qualtran/bloqs/factoring/mod_mul.py]: No, this was my first attempt, but the root problem was that the override in bloq_example was missing a default.  I missed reverting these after I discovered that.

File: qualtran/bloqs/swap_network/cswap_approx_test.py (hunk 3 above)
Comment (dstrain115) [reply qualtran/bloqs/swap_network/cswap_approx_test.py:51]: Good to know.  I will use that in the future.  Yeah, that type is fairly well hidden.

File: qualtran/bloqs/factoring/mod_mul.py (hunk 2 above)
Comment (mpharrigan) [reply qualtran/bloqs/factoring/mod_mul.py]: cool, that's what I was hoping you'd say :)

**Review by mpharrigan — APPROVED** [review]
groovy

**Comment by mpharrigan** [comment]
Looks like a legitimate conflict with #867 -- which was just merged.

**Pull request merged**